### PR TITLE
Typescriptize assetUtils file

### PR DESCRIPTION
**Pull request by magicznyleszek**
## Description

Adds types to assetUtils file.

## Related issues

Contains the work from these PRs:

- #3429
- #3435

### CODE REVIEW

File: jsapp/js/dataInterface.d.ts [hunk 1]
```diff
@@ -242,3 +262,51 @@ interface AssetResponse {
   settings__form_id?: string
   settings__title?: string
 }
+
+interface PaginatedResponse {
+  count: number
+  next: null | string
+  previous: null | string
+  results: any[]
```
Comment (duvld) [inline jsapp/js/dataInterface.d.ts:270]: Curious if a `Submission` type is a todo, or there's another reason why this is `any`?

File: jsapp/js/assetUtils.ts [hunk 2]
```diff
@@ -394,45 +396,63 @@ export function isRowSpecialLabelHolder(mainRow, holderRow) {
...
-  let typeDef;
-  if (type === SCORE_ROW_TYPE) {
-    typeDef = QUESTION_TYPES.score;
-  } else if (type === RANK_LEVEL_TYPE) {
-    typeDef = QUESTION_TYPES.rank;
-  } else {
-    typeDef = QUESTION_TYPES[type];
+export function renderQuestionTypeIcon(
+  rowType: AnyRowTypeName
+): React.DetailedReactHTMLElement<{}, HTMLElement> | null {
```
Comment (duvld) [inline jsapp/js/assetUtils.ts:413]: Still new to some syntax, so I want to make sure I understand this generic correctly:
` React.DetailedReactHTMLElement<{}, HTMLElement>` is an object that takes in an HTML element and some object as parameters for that HTML element?

File: jsapp/js/components/permissions/permConfig.ts [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+   * Throws if trying to use permConfig before it fetches data from BE or if
+   * data given by `setPermissions` is not an array.
+   */
+  verifyReady() {
+    if (!this.isReady()) {
+      throw new Error(t('Permission config is not ready or failed to initialize!'));
+    }
+  }
+
+  public isReady() {
```
[the remark this answers is not in the document]
Comment (magicznyleszek) [reply jsapp/js/components/permissions/permConfig.ts:30]: Yeah, `verifyReady` is supposed to be private. Fixed that :)

File: jsapp/js/dataInterface.d.ts [hunk 4]
```diff
@@ -75,13 +52,54 @@ interface ExportSetting {
...
+// TODO: most of these could be changed into ENUMS from just strings (e.g. `type`)
+/**
+ * It represents a question from the form, a group start/end or a piece of
+ * a more complex question type.
+ */
+interface SurveyRow {
+  $autoname: string
+  $kuid: string
+  // We use dynamic import to avoid changing this ambient module to a normal
+  // module: see https://stackoverflow.com/a/51114250/2311247
```
[the remark this answers is not in the document]
Comment (magicznyleszek) [reply jsapp/js/dataInterface.d.ts:63]: Took me a lot of time to figure this one out 😅

File: jsapp/js/assetUtils.ts (hunk 2 above)
Comment (magicznyleszek) [reply jsapp/js/assetUtils.ts:413]: I think the object is for custom node attributes. I don't fully get the `<>` in types yet. I'm often just looking out what people do and looking through the types definitions of our packages :)

**Pull request merged**